libbeat/tests/system/test_keystore.py

```diff
@@ -58,17 +58,17 @@ def test_keystore_with_nested_key(self):
         key = "output.elasticsearch.hosts.0"
         secret = "myeleasticsearchsecrethost"
 
-        self.render_config_template(keystore_path=self.keystore_path, elasticsearch={
-            'hosts': "${%s}" % key
-        })
+        self.render_config_template("mockbeat",
+                                    keystore_path=self.keystore_path,
+                                    output_file_path="${%s}" % key)
 
-        exit_code = self.run_beat(extra_args=["keystore", "create"])
+        exit_code = self.run_beat(extra_args=["keystore", "create"],
+                                  config="mockbeat.yml")
         assert exit_code == 0
 
         self.add_secret(key, secret)
-        proc = self.start_beat()
-        self.wait_until(lambda: self.log_contains("Elasticsearch url: http://myeleasticsearchsecrethost:9200"))
-        assert self.log_contains(secret)
```

Comment:
We don't check for the secret now?

Reply:
I could check the creating of the path, I though it would still rely on IO and might be flaky, instead we rely on the access of the key in the keystore, when that log statement appear the secret is retrieved.

Reply:
@jsoriano I changed the assert, I look for the existence of the directory, and the name is coming from the keystore. To mitigate the timing issue, I assert the ACK on the event before shutting down the beat. When the ACK is sent from the output, the event should be persisted on disk, and the directory will exist.